### DISCUSSION TITLE
How to determine which version of Typescript Bun is relying on?

### DISCUSSION THREAD

**Discussion by guest271314** [oven-sh/bun#14758] [Q&A] [answered]
How to determine which version of Typescript Bun is relying on?

**Comment by guest271314** [comment]
Is Bun using `swc` for TypeScript compilation and type stripping?

**Reply by Jarred-Sumner** [reply, marked as answer]
Bun has it's own javascript parser that supports TypeScript and JSX. Most of the code is in this file https://github.com/oven-sh/bun/blob/main/src/js_parser.zig#L1

**Comment by JJ** [comment]
I'm sorry to enliven this thread again, but this does not really answer the question. I can interpret is as meaning that there's no fixed Typescript version in bun, but more a general support, with specific version support delegated to `@types/bun`; for instance, [a change in @types/bun was needed to fix support for TS 5.9](https://github.com/oven-sh/bun/issues/21535).
[@types/bun is tagged with the TS version it supports](https://www.npmjs.com/package/@types/bun?activeTab=versions), so I guess the answer to this would be that `bun` is relying on whatever TS version @types/bun is tagged with. `1.3.8`, for instance, is tagget `6.0`, so there you are. Changing TS versions is, thus, a matter of finding the corresponding `@bun/types` version and installing it.

**Reply by vpedrosa** [reply]
I've been testing this and I think there's an important nuance. @types/bun only provides type definitions — it affects tsc and IDE autocompletion, but not Bun's own transpiler.
                                                                                                                                                                                              
I tested this with a [TS 5.8 feature](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-8.html) (stricter type checking in conditional return expressions):                                                                                             
```typescript
  declare const untypedCache: Map<any, any>;
  function getUrlObject(urlString: string): URL {
      return untypedCache.has(urlString) ?
          untypedCache.get(urlString) :
          urlString;  // should error: Type 'string' is not assignable to type 'URL'
  }
```
  - bun build / bun run (v1.3.5) without @types/bun: no error
  - bun build / bun run (v1.3.5) with @types/bun@1.3.8: no error
  - tsc 5.8 --noEmit: error detected

So bun build just transpiles — it doesn't do type checking regardless of what @types/bun is installed. Changing @types/bun would only matter if you use tsc for type checking. In that case, yes, you'd need a @types/bun version compatible with your typescript version, but that's a tsc concern, not a Bun runtime concern.

**Reply by JJ** [reply]
Boils down to: not entirely clear what version of Typescript is included in every version of bun?
